app/controllers/ultima_sample_sheet/sample_sheet_generator.rb

```diff
@@ -0,0 +1,232 @@
+# frozen_string_literal: true
+
+require 'zip'
+
+module UltimaSampleSheet::SampleSheetGenerator
+  # Initiates the sample sheet generation for the given batch.
+  # @param batch [UltimaSequencingBatch] the batch to generate sample sheets for
+  # @return [String] the ZIP archive as a binary string
+  def self.generate(batch)
+    Generator.new(batch).generate
+  end
+
+  # Ultima sample sheet generator class.
+  # It creates a ZIP archive containing individual sample sheet CSV files
+  # for each request in the given Ultima sequencing batch.
+  class Generator
+    PLATE_LENGTH = 8 # Assumes 96-well tag plates with 8 rows (A-H).
+    HEADER_TITLE = ['[Header]'].freeze
+    GLOBAL_TITLE = ['[Global]'].freeze
+    GLOBAL_HEADERS = %w[
+      Application
+      sequencing_recipe
+      analysis_recipe
+    ].freeze
+    SAMPLES_TITLE = ['[Samples]'].freeze
+    SAMPLES_HEADERS = %w[
+      Sample_ID
+      Library_name
+      Index_Barcode_Num
+      Index_Barcode_Sequence
+      Barcode_Plate_Num
+      Barcode_Plate_Well
+      application_type
+      study_id
```

Comment:
Are we allowed to add extra / non standard headers  (study_id), have we checked whether ultima accepts them?

Reply:
Additional columns are allowed by 

spec [UG Sample Sheet Reference Guide](https://cdn.sanity.io/files/l7780ks7/production-2024/4ca366aea0f7ac74a70c39897f870479ddeaf3dc.pdf) Section: Examples of Sample Sheets using Optional Fields 

and existing sample sheet examples Confluence [[Ultima UG 100 Sequencer](https://ssg-confluence.internal.sanger.ac.uk/spaces/PSDPUB/pages/271057969/Ultima+UG+100+Sequencer)] Section Ultima Sample Sheets